pkg/ingester/limiter.go

```diff
@@ -210,10 +210,6 @@ func (l *Limiter) convertGlobalToLocalLimit(userID string, globalLimit int) int
 		return 0
 	}
 
-	// Given we don't need a super accurate count (ie. when the ingesters
-	// topology changes) and we prefer to always be in favor of the tenant,
-	// we can use a per-ingester limit equal to:
-	// (global limit / number of ingesters) * replication factor
 	numIngesters := l.ring.HealthyInstancesCount()
```

Comment:
I wouldn't read it anymore, but use the value of `getHealthyIngestersInZoneCount()`. If it's 0, then we return `0` from here. Also to avoid race conditions, we should read `getHealthyIngestersInZoneCount()` only once and then use that value in the rest of the function.